Logging of OC caches crash the app

Sorry for formatting errors...posting from mobile

2020.09.10-NB

Sequence:
- Open OC14364
- Select Menu - Log Visit (in my case I used Fast offline log - Log visit if it makes a difference)
- Crash

Log:

```
--------- beginning of crash
09-10 21:10:09.263  1632  1632 E AndroidRuntime: FATAL EXCEPTION: main
09-10 21:10:09.263  1632  1632 E AndroidRuntime: Process: cgeo.geocaching, PID: 1632
09-10 21:10:09.263  1632  1632 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{cgeo.geocaching/cgeo.geocaching.log.LogCacheActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'boolean cgeo.geocaching.connector.capability.IVotingCapability.isValidRating(float)' on a null object reference
09-10 21:10:09.263  1632  1632 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3654)
09-10 21:10:09.263  1632  1632 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3806)
09-10 21:10:09.263  1632  1632 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
09-10 21:10:09.263  1632  1632 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
09-10 21:10:09.263  1632  1632 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
09-10 21:10:09.263  1632  1632 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2267)
09-10 21:10:09.263  1632  1632 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:107)
09-10 21:10:09.263  1632  1632 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:237)
09-10 21:10:09.263  1632  1632 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8154)
09-10 21:10:09.263  1632  1632 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
09-10 21:10:09.263  1632  1632 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
09-10 21:10:09.263  1632  1632 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
09-10 21:10:09.263  1632  1632 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'boolean cgeo.geocaching.connector.capability.IVotingCapability.isValidRating(float)' on a null object reference
09-10 21:10:09.263  1632  1632 E AndroidRuntime: 	at cgeo.geocaching.ui.CacheVotingBar.setRating(CacheVotingBar.java:53)
09-10 21:10:09.263  1632  1632 E AndroidRuntime: 	at cgeo.geocaching.log.LogCacheActivity.resetValues(LogCacheActivity.java:418)
09-10 21:10:09.263  1632  1632 E AndroidRuntime: 	at cgeo.geocaching.log.LogCacheActivity.onCreate(LogCacheActivity.java:298)
09-10 21:10:09.263  1632  1632 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:7963)
09-10 21:10:09.263  1632  1632 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:7952)
09-10 21:10:09.263  1632  1632 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1307)
09-10 21:10:09.263  1632  1632 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3629)
09-10 21:10:09.263  1632  1632 E AndroidRuntime: 	... 11 more
09-10 21:10:09.387 15743 15743 E cgeo.geocachin: Not starting debugger since process cannot load the jdwp agent.
09-10 21:10:09.414 15743 15743 I cgeo.geocachin: The ClassLoaderContext is a special shared library.
09-10 21:10:09.442 15743 15743 W cgeo.geocachin: Accessing hidden field Landroid/view/ViewConfiguration;->sHasPermanentMenuKey:Z (greylist-max-p, reflection, denied)
09-10 21:10:09.445 15743 15743 I cgeo    : [Log] Logging set: minLevel=WARN, minAddCallerInfo=NONE, throwOnError=false
09-10 21:10:09.445 15743 15743 I cgeo    : [Log] No logging config found at /storage/emulated/0/cgeo/logfiles/log.properties, using defaults
09-10 21:10:09.445 15743 15743 I cgeo    : [Log] Logging set: minLevel=WARN, minAddCallerInfo=NONE, throwOnError=false
09-10 21:10:14.905 15743 15743 E AndroidRuntime: FATAL EXCEPTION: main
09-10 21:10:14.905 15743 15743 E AndroidRuntime: Process: cgeo.geocaching, PID: 15743
09-10 21:10:14.905 15743 15743 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{cgeo.geocaching/cgeo.geocaching.log.LogCacheActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'boolean cgeo.geocaching.connector.capability.IVotingCapability.isValidRating(float)' on a null object reference
09-10 21:10:14.905 15743 15743 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3654)
09-10 21:10:14.905 15743 15743 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3806)
09-10 21:10:14.905 15743 15743 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
09-10 21:10:14.905 15743 15743 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
09-10 21:10:14.905 15743 15743 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
09-10 21:10:14.905 15743 15743 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2267)
09-10 21:10:14.905 15743 15743 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:107)
09-10 21:10:14.905 15743 15743 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:237)
09-10 21:10:14.905 15743 15743 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8154)
09-10 21:10:14.905 15743 15743 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
09-10 21:10:14.905 15743 15743 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
09-10 21:10:14.905 15743 15743 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
09-10 21:10:14.905 15743 15743 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'boolean cgeo.geocaching.connector.capability.IVotingCapability.isValidRating(float)' on a null object reference
09-10 21:10:14.905 15743 15743 E AndroidRuntime: 	at cgeo.geocaching.ui.CacheVotingBar.setRating(CacheVotingBar.java:53)
09-10 21:10:14.905 15743 15743 E AndroidRuntime: 	at cgeo.geocaching.log.LogCacheActivity.resetValues(LogCacheActivity.java:418)
09-10 21:10:14.905 15743 15743 E AndroidRuntime: 	at cgeo.geocaching.log.LogCacheActivity.onCreate(LogCacheActivity.java:298)
09-10 21:10:14.905 15743 15743 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:7963)
09-10 21:10:14.905 15743 15743 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:7952)
09-10 21:10:14.905 15743 15743 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1307)
09-10 21:10:14.905 15743 15743 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3629)
09-10 21:10:14.905 15743 15743 E AndroidRuntime: 	... 11 more
09-10 21:10:17.441 16003 16003 E cgeo.geocachin: Not starting debugger since process cannot load the jdwp agent.
09-10 21:10:17.461 16003 16003 I cgeo.geocachin: The ClassLoaderContext is a special shared library.
09-10 21:10:17.482 16003 16003 W cgeo.geocachin: Accessing hidden field Landroid/view/ViewConfiguration;->sHasPermanentMenuKey:Z (greylist-max-p, reflection, denied)
09-10 21:10:17.484 16003 16003 I cgeo    : [Log] Logging set: minLevel=WARN, minAddCallerInfo=NONE, throwOnError=false
09-10 21:10:17.485 16003 16003 I cgeo    : [Log] No logging config found at /storage/emulated/0/cgeo/logfiles/log.properties, using defaults
09-10 21:10:17.485 16003 16003 I cgeo    : [Log] Logging set: minLevel=WARN, minAddCallerInfo=NONE, throwOnError=false
09-10 21:10:38.424 16003 16003 E AndroidRuntime: FATAL EXCEPTION: main
09-10 21:10:38.424 16003 16003 E AndroidRuntime: Process: cgeo.geocaching, PID: 16003
09-10 21:10:38.424 16003 16003 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{cgeo.geocaching/cgeo.geocaching.log.LogCacheActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'boolean cgeo.geocaching.connector.capability.IVotingCapability.isValidRating(float)' on a null object reference
09-10 21:10:38.424 16003 16003 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3654)
09-10 21:10:38.424 16003 16003 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3806)
09-10 21:10:38.424 16003 16003 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
09-10 21:10:38.424 16003 16003 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
09-10 21:10:38.424 16003 16003 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
09-10 21:10:38.424 16003 16003 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2267)
09-10 21:10:38.424 16003 16003 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:107)
09-10 21:10:38.424 16003 16003 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:237)
09-10 21:10:38.424 16003 16003 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8154)
09-10 21:10:38.424 16003 16003 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
09-10 21:10:38.424 16003 16003 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
09-10 21:10:38.424 16003 16003 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
09-10 21:10:38.424 16003 16003 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'boolean cgeo.geocaching.connector.capability.IVotingCapability.isValidRating(float)' on a null object reference
09-10 21:10:38.424 16003 16003 E AndroidRuntime: 	at cgeo.geocaching.ui.CacheVotingBar.setRating(CacheVotingBar.java:53)
09-10 21:10:38.424 16003 16003 E AndroidRuntime: 	at cgeo.geocaching.log.LogCacheActivity.resetValues(LogCacheActivity.java:418)
09-10 21:10:38.424 16003 16003 E AndroidRuntime: 	at cgeo.geocaching.log.LogCacheActivity.onCreate(LogCacheActivity.java:298)
09-10 21:10:38.424 16003 16003 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:7963)
09-10 21:10:38.424 16003 16003 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:7952)
09-10 21:10:38.424 16003 16003 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1307)
09-10 21:10:38.424 16003 16003 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3629)
09-10 21:10:38.424 16003 16003 E AndroidRuntime: 	... 11 more
09-10 21:10:41.783 16706 16706 E cgeo.geocachin: Not starting debugger since process cannot load the jdwp agent.
09-10 21:10:41.800 16706 16706 I cgeo.geocachin: The ClassLoaderContext is a special shared library.
09-10 21:10:41.822 16706 16706 W cgeo.geocachin: Accessing hidden field Landroid/view/ViewConfiguration;->sHasPermanentMenuKey:Z (greylist-max-p, reflection, denied)
09-10 21:10:41.825 16706 16706 I cgeo    : [Log] Logging set: minLevel=WARN, minAddCallerInfo=NONE, throwOnError=false
09-10 21:10:41.825 16706 16706 I cgeo    : [Log] No logging config found at /storage/emulated/0/cgeo/logfiles/log.properties, using defaults
09-10 21:10:41.825 16706 16706 I cgeo    : [Log] Logging set: minLevel=WARN, minAddCallerInfo=NONE, throwOnError=false

```

--- a/main/src/cgeo/geocaching/ui/CacheVotingBar.java
+++ b/main/src/cgeo/geocaching/ui/CacheVotingBar.java
@@ -50,9 +50,9 @@ public class CacheVotingBar {
     }
 
     public void setRating(final float stars) {
-        final float rating = votingConnector.isValidRating(stars) ? stars : 0;
+        final float rating = votingConnector != null && votingConnector.isValidRating(stars) ? stars : 0;
         ratingBar.setRating(rating);
-        label.setText(votingConnector.getDescription(rating));
+        label.setText(votingConnector == null ? "--" : votingConnector.getDescription(rating));
         this.rating = rating;
         if (changeListener != null) {
             changeListener.onRatingChanged(rating);

